File: TransCoder_tokenized_test_set_functions/cpp/COUNT_OF_OCCURRENCES_OF_A_101_PATTERN_IN_A_STRING.cpp
int countPattern ( string str ) {
  int len = str . size ( );
  bool oneSeen = 0;
  int count = 0;
  for ( int i = 0;
  i < len;
  i ++ ) {
    if ( str [ i ] == '1' && oneSeen == 1 ) if ( str [ i - 1 ] == '0' ) count ++;
    if ( str [ i ] == '1' && oneSeen == 0 ) {
      oneSeen = 1;
      continue;
    }
    if ( str [ i ] != '0' && str [ i ] != '1' ) oneSeen = 0;
  }
  return count;
}

File: results/java_codex/COUNT_OF_OCCURRENCES_OF_A_101_PATTERN_IN_A_STRING.java
int countPattern ( string str ) {
  int len = str . length ( );
  bool oneSeen = false;
  int count = 0;
  for ( int i = 0;
  i < len;
  i ++ ) {
    char getChar = str [ i ];
    if ( getChar == '1' && oneSeen == true ) {
      if ( str [ i - 1 ] == '0' ) count ++;
    }
    if ( getChar == '1' && oneSeen == false ) oneSeen = true;
    if ( getChar != '0' && str [ i ] != '1' ) oneSeen = false;
  }
  return count;
}